frontend/OBSApp.cpp

```diff
@@ -1229,7 +1285,20 @@ bool OBSApp::OBSInit()
 	mainWindow = new OBSBasic();
 
 	mainWindow->setAttribute(Qt::WA_DeleteOnClose, true);
-	connect(mainWindow, &OBSBasic::destroyed, this, &OBSApp::quit);
+
+	connect(QApplication::instance(), &QApplication::aboutToQuit, this, [this]() {
+		crashHandler_->applicationShutdownHandler();
+
+		/* Ensure OBSMainWindow gets closed */
+		if (mainWindow) {
+			mainWindow->close();
+			delete mainWindow;
+		}
+
+		if (libobs_initialized) {
+			applicationShutdown();
+		}
```

Comment:
Should be fine, but also should be noted that this is necessary only because `applicationShutdown` is now moved into the `aboutToQuit` handler

Otherwise `QCoreApplication::exit` would have emitted the event, closed and deleted the main window, and `OBSApp`'s destructor would have run last.